bears/vcs/git/GitCommitBear.py

```diff
@@ -24,4 +25,33 @@ def get_remotes():
 
     def get_head_commit(self):
         with change_directory(self.get_config_dir() or os.getcwd()):
+            command = self.github_pull_request_temporary_merge_commit()
+            if command:
+                return run_shell_command(command)
+
             return run_shell_command('git log -1 --pretty=%B')
+
+    def github_pull_request_temporary_merge_commit(self):
+        """
+        This detects the merge commit generated by GitHub
+        in a refs/pull/(\d+)/merge git remote reference.
+        Visit https://github.com/travis-ci/travis-ci/issues/8400
+        for more details.
```

Comment:
Also add in the docstring about what thing will this function return, what will it return if it found the github temporary pull request commit, and what will it return if it doesn't. Use `:return:` style